external/xtorch_harmonics/tests/test_roundtrip.py

```diff
@@ -112,25 +121,56 @@ def test_roundtrip_dataarray_preserves_dtype(dtype):
         (LOBATTO_GRID, 1e-5),
     ],
 )
-def test_roundtrip_real_spherical_harmonic_dataarray(grid, rtol):
+def test_roundtrip_filter_real_spherical_harmonic_dataarray(grid, rtol):
     # We expect spherical harmonics themselves to pass through approximately
     # unchanged if we use the same grid for the forward and inverse transforms.
-    # This also tests the use of roundtrip on DataArrays with more than two
+    # This also tests the use of roundtrip_filter on DataArrays with more than two
     # dimensions; da has dimensions ["grid_yt", "grid_xt", "m", "n"].
-    da = real_spherical_harmonic_dataarray(grid, LAT_DIM, LON_DIM)
-    result = roundtrip(da, LAT_DIM, LON_DIM, forward_grid=grid, inverse_grid=grid)
+    lat, lon = grid_data_arrays(grid, LAT_DIM, LON_DIM)
+    da = real_spherical_harmonic_dataarray(lat, lon)
+    result = roundtrip_filter(
+        da, LAT_DIM, LON_DIM, forward_grid=grid, inverse_grid=grid
+    )
+    print(da)
+    print(result.compute())
     xr.testing.assert_allclose(result, da, rtol=rtol)
 
 
+@pytest.mark.parametrize(("fraction_modes_kept"), [None, 0.5, 0.75, 1.0])
+def test_roundtrip_filter_truncation_real_spherical_harmonic_dataarray(
+    fraction_modes_kept, grid=LEGENDRE_GAUSS_GRID
```

Comment:
Nice test.  

nit: if not parameterizing over `grid` maybe move it to the function body.

I'm guessing the reason you do not parametrize over `grid` is that the other grids need a coarser tolerance for the test to pass.  As you probably saw, I was also somewhat puzzled by this earlier, though it was not immediately clear what could cause the difference other than some detail in `torch_harmonics` or the numerics themselves.